src/engine/plugins/tile.cpp

```diff
@@ -400,206 +379,197 @@ Status TilePlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataFacad
-                            }
-
-                            // Add the turn angle value to the value lookup table for the vector
-                            // tile.
-                            const auto turn_angle_index = use_point_int_value(turn_angle);
-                            // And, same for the actual turn cost value - it goes in the lookup
-                            // table,
-                            // not directly on the feature itself.
-                            const auto turn_cost_index = use_point_float_value(
-                                turn_cost / 10.0); // Note conversion to float here
-
-                            // Save everything we need to later add all the points to the tile.
-                            // We need the coordinate of the intersection, the angle in, the turn
-                            // angle and the turn cost.
-                            all_turn_data.emplace_back(TurnData{
-                                coord_via, angle_in_index, turn_angle_index, turn_cost_index});
+                            approach_weight_vector = facade->GetUncompressedReverseWeights(
+                                edge_based_node_info[approachedge.edge_based_node_id]
+                                    .packed_geometry_id);
                         }
+                        const auto sum_node_weight = std::accumulate(approach_weight_vector.begin(),
+                                                                     approach_weight_vector.end(),
+                                                                     EdgeWeight{0});
+
+                        // The edge.weight is the whole edge weight, which includes the turn
+                        // cost.
+                        // The turn cost is the edge.weight minus the sum of the individual road
+                        // segment weights.  This might not be 100% accurate, because some
+                        // intersections include stop signs, traffic signals and other
+                        // penalties, but at this stage, we can't divide those out, so we just
+                        // treat the whole lot as the "turn cost" that we'll stick on the map.
+                        const auto turn_cost = data.weight - sum_node_weight;
+
+                        // Find the three nodes that make up the turn movement)
+                        const auto node_from = startnode.first;
+                        const auto node_via = approachedge.target_node;
+                        const auto node_to = exit_edge.target_node;
+
+                        const auto coord_from = facade->GetCoordinateOfNode(node_from);
+                        const auto coord_via = facade->GetCoordinateOfNode(node_via);
+                        const auto coord_to = facade->GetCoordinateOfNode(node_to);
+
+                        // Calculate the bearing that we approach the intersection at
+                        const auto angle_in = static_cast<int>(
```

Comment:
These turn angles are for map rendering - they need to align with the exact visual geometry so that icons can be placed next to the lines on the map.

Exposing the guidance angles as part of the tiles could be useful, but I think we should do that later if we decide we need them.  I think if we tried to use them for a visual layer, it might be difficult to determine which turn arrows